lib/gnat/consumer_supervisor.ex

```diff
@@ -84,23 +79,15 @@ defmodule Gnat.ConsumerSupervisor do
       status: :disconnected,
       subscription_topics: subscription_topics,
       subscriptions: [],
-      task_supervisor_pid: task_supervisor_pid,
-      service_config: microservice,
+      task_supervisor_pid: task_supervisor_pid
     }
 
-    cond do
-      Map.has_key?(settings, :module) ->
-        state = Map.put(state, :module, settings.module)
-        send self(), :connect
-        {:ok, state}
-
-      Map.has_key?(settings, :consuming_function) ->
-        state = Map.put(state, :consuming_function, settings.consuming_function)
-        send self(), :connect
-        {:ok, state}
-
-      true ->
-        {:error, "You must provide a module or consuming function for the consumer supervisor"}
+    with {:ok, state} <- maybe_append_service(state, settings),
```

Comment:
Curious if you'll like this change. The `cond` block was a bit tricky to read and I wanted to validate the user-input about the service definition during process startup rather than crashing the process later when the connection became available.

So I used a `with` block here to show the happy path, and I perform all of the service configuration validation right here (as well as initializing service counters etc). This way our stats will persist across re-connects and if the user has an invalid validation they'll get a helpful error at boot time rather than a process crash later